Hardware location update

joachimweyl opened the pull request:
update row, pod, and cage for all of the hosts. Add the unitnumber
within the cage to infra and prod hosts. Will add test when we
get that information

joachimweyl committed:
Update hardware location info minus elevation

We want the hardware.csv to house location information.
I already had the row, pod, cage/rack information.
I will add the unitnumber informaiton once I get it.
Part-of #2

joachimweyl committed:
Add unitnumbers for hosts minus test

Include unitnumbers for fc430 hosts in the fx2 chassis so that we
know where in the cage they are. Also included unitnumbers for
the sd530 hosts in the thinksystem chassis. We just need the
r640 hosts and the r515 hosts for the test cluster.

mikthoma reviewed (approved):
Looks good!

joachimweyl committed:
Merge branch 'main' into HardwareLocation

joachimweyl commented:
@larsks I used rebase to clean up my commits. Then had to force push to my repo. Is that what is causing the inability to merge? I did not change any files that had been changed by the 1 other merge you did on the public repo so there should not be any conflicts.

larsks commented:
@joachimweyl you don't have merge privileges on this repository at the moment. I can update the permissions if you want, but we can change that if you'd like.

larsks commented on inventory/00-static/ocp-infra.yaml:
```diff
@@ -12,5 +12,17 @@ all:
+                  row: 07
+                  pod: C
+                  cage: 03
+                  unitnumber: 09-10
```
I would prefer `unitnumber` to be a single value (typically, the lowest position the device occupies). If you want to record the device height, let's add a `height` parameter (but I'm not sure we need that at the moment; I guess I would be inclined to leave it out for now). Also, be careful writing numbers with leading zeroes: I don't think it's a problem here, but some tools interpret a leading 0 to mean "this is an octal number". If these are supposed to be numeric, strip the leading zero. If they are meant to be an opaque identifier, explicitly quote them (`cage: "03"`).

joachimweyl replied:
Will do.

joachimweyl replied:
@larsks updated with no leading zero's and unit numbers only including the lower number.

joachimweyl commented:
> @joachimweyl you don't have merge privileges on this repository at the moment. I can update the permissions if you want, but we can change that if you'd like.

Got it, so it is not an issue with the PR, good. No need to change.

joachimweyl committed:
Update hardware info to fix formatting

Remove leading zero and only provide lowest number for
the unitnumber. We can add height later if we feel we need
this information.

joachimweyl committed:
Merge branch 'HardwareLocation' of github.com:joachimweyl/nerc-ansible into HardwareLocation

larsks commented:
@joachimweyl I've pushed a minor update that makes all the location metadata attributes of a top-level `location` variable, so:

```
              hosts:
                ctl-0.nerc-ocp-infra.rc.fas.harvard.edu:
                  location:
                    row: 7
                    pod: C
                    cage: 3
                    unit: 9
```

That makes it a little more convenient to use programatically. Other than that, looks good. If you're okay with this change I'll go ahead and merge it.

larsks reviewed (approved):
👍
